### ISSUE TITLE
Stuck in wrong state when text added to an empty YoshikoTextField programatically

### ISSUE THREAD

**Issue by balazsgerlei** [raulriera/TextFieldEffects#57]
When the TextField's text property is set in viewDidAppear, the "effect" is not shown, and the hint is displayed behind the actual text. The proplem don't happen when the text is set in viewDidLoad though.

**Comment by raulriera** [comment]
Hey, did you find a solution for this?

**Comment by balazsgerlei** [comment]
Hi!

The thing is that I realized that what I wrote is not the culprit. The problem is still there, I only closed the issue because I wanted to find the real cause and then reopen it or open a new one accordingly.

Anyway, this is what I currently know:
I have a screen where there are severeal ViewControllers embedded in a TabBarController. In the TabBarController I initiate a network request to sync local data with the backend (asynchron call on a background thread). When the data arrives I display the data in the child viewcontrollers of the TabBarController (only with the one currently showing). But when a user switches to a tab after the data loaded, the data won't be there, so I also display it in the viewDidLoad method (this was done in the viewDidAppear previously).

One of the child ViewControllers is a "Profile" screen, where I have several UILabels that display the user's details. I also have TextFields (YoshikoTextFields from this library) for editing these. They are hidden by default (from the storyboard) and only appear when the user taps the "Edit" button. I update the labels and the textfields when the data arrives from the server and in viewDidLoad, but in the storyboard they have no text.

I previously thought that the problem happens when the text property is set in the viewDidAppear, but no, it happens when the profile screen is the selected item in the tab bar, and the data arrives from the server. There is a closure in the TabBarController which handles the result of the server call, calls the refresh method in the profile screen's viewcontroller which than sets the text property of the labels and textfields (for which user details are present). At this stage the TextFields are still hidden. When I (or the users) tap edit, they appear in a wrong visual stata: the text I set previously is entered but the hint is still displayed behind them, and not above. It olnly goes back to normal when I delete the text, tap to another field and than back to the textfield in question.

This is what I know currently, the temporary workaround was that I put a single space character as text in the storyboard and clear it when the data arrives and there is no user detail to be displayed in a particular textfield. I'm still investigating the problem though. If you have any suggestions, I would appreciate it! Thanks!

**Comment by raulriera** [comment]
Hi,

Are you sure you are updating the textfield in the main queue? Also, if you have code would be way better to debug.

**Comment by balazsgerlei** [comment]
I also found out more. It happens when I set the text property of the textfield from code, and the text field is visible AND it has no text. I updated the title and reopened the issue.

**Comment by raulriera** [comment]
If you are feeling adventurous, the problem lies in these 3 methods :dancers: 

```swift
private func animateViews() {
        UIView.animateWithDuration(0.2, animations: {
            self.placeholderLabel.alpha = 0
            self.placeholderLabel.frame = self.placeholderRectForBounds(self.bounds)
            
            }) { complete in
                self.updatePlaceholder()
                UIView.animateWithDuration(0.3, animations: {
                    self.placeholderLabel.alpha = 1
                    self.updateBorder()
                    self.updateBackground()
                })
        }
    }
    
    // MARK: - TextFieldEffects
    
    override public func animateViewsForTextEntry() {
        guard text!.isEmpty else { return }
        
        animateViews()
    }
    
    override public func animateViewsForTextDisplay() {
        guard text!.isEmpty else { return }
        
        animateViews()
    }
```

**Comment by raulriera** [comment]
Fixed in version 1.1.1

**Comment by balazsgerlei** [comment]
Awesome! Thank you! What was the problem? I wanted to help you find the bug, but I didn't have time too look at it.
